src/Build/Graph/ProjectInterpretation.cs

```diff
@@ -115,7 +115,12 @@ internal static ProjectType GetProjectType(ProjectInstance project)
                     : ProjectType.NonMultitargeting;
         }
 
-        public static void PostProcess(ConcurrentDictionary<ConfigurationMetadata, ProjectGraphNode> allNodes)
+        /// <summary>
+        /// To avoid calling nuget at graph construction time, the graph is initially constructed with outer build nodes referencing inner build nodes.
+        /// However, at build time, for non root outer builds, the inner builds are NOT referenced by the outer build, but by the nodes referencing the
+        /// outer build. Change the graph to mimic this behaviour.
```

Comment:
I don't understand. It sounds like "it starts outer->inner, but at build time it's actually outer->inner", but that doesn't make sense. Can you rephrase?

Reply:
Updated comment with example.